cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGeneratorTest.java

```diff
@@ -120,6 +140,44 @@ public void echo(HttpServiceRequest request, HttpServiceResponder responder) {
     }
   }
 
+  /**
+   * A testing handler for testing file upload through usage of {@link HttpContentConsumer}.
+   */
+  public static final class FileUploadHandler extends AbstractHttpServiceHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FileUploadHandler.class);
+    private final File outputDir;
+
+    public FileUploadHandler(File outputDir) {
+      this.outputDir = outputDir;
+    }
+
+    @Path("/upload/{file}")
+    @PUT
+    public HttpContentConsumer upload(HttpServiceRequest request,
+                                      HttpServiceResponder responder,
+                                      @PathParam("file") String file) throws IOException {
+      final WritableByteChannel channel = Channels.newChannel(new FileOutputStream(new File(outputDir, file)));
+      return new HttpContentConsumer() {
+        @Override
+        public void onReceived(ByteBuffer chunk) throws Exception {
+          channel.write(chunk);
+        }
+
+        @Override
+        public void onFinish(HttpServiceResponder responder) throws Exception {
+          channel.close();
+          responder.sendStatus(200);
+        }
+
+        @Override
```

Comment:
In this test, no. There is another test that covers failure upload case. This test is more for testing the byte code generation correctness.